release/BUILD

```diff
@@ -27,6 +27,20 @@ pkg_tar(
     ],
 )
 
+pkg_tar(
+    name = "kubernetes-node-windows-amd64",
+    srcs = [
+        "//cmd/auth-provider-gcp",
+    ],
+    extension = "tar.gz",
+    remap_paths = {
+        "/auth-provider-gcp.exe": "kubernetes/node/bin/auth-provider-gcp.exe",
```

Comment:
"auth"-provider seems slightly misnamed. In other places we refer to credential-provider. If we are trying to disambiguate from the sig-auth credential provider, I'm not sure this would be a great choice in name.

Reply:
I think `credential-provider` would be better as well, though unfortunately the Linux deployment has a similar issue (see [here](https://github.com/kubernetes/cloud-provider-gcp/blob/bab0156ad51c5cb24e60f7553d552ac34ff09544/release/BUILD#L19)) - since this affects Linux as well, I think a quick issue + followup PR to rename where relevant on both platforms (the `BUILD` file for the credential provider, any relevant deployment scripts, etc.).

Another way would be to do a partial fix via `remap_paths`, i.e, `"/auth-provider-gcp.exe": "kubernetes/node/bin/credential-provider.exe"` in this PR and then follow up with the other solution above, but I'm uncertain how much extra value that adds.